src/components/shared/Rating/index.tsx

```diff
@@ -59,80 +45,46 @@ const StyledImage = styled.Image`
 `;
 
 function StarComponent(props: StarProps): React.ReactElement {
-  const [focus, setfocus] = useState(false);
-
-  const _handlePressIn = (): void => {
-    setfocus(true);
-  };
-
   const handlePress = (): void => {
     props.onPress();
   };
 
-  const _handlePressOut = (): void => {
-    setfocus(false);
-  };
-
   const image = props.on
     ? require('./images/Star.png')
     : require('./images/UnStar.png');
 
   return (
-    <StarWrapper
-      focus={focus}
-      onPressIn={_handlePressIn}
-      onPress={handlePress}
-      onPressOut={_handlePressOut}
-      activeOpacity={props.disabled ? 1 : 0.7}
-    >
+    <StarWrapper onPress={handlePress} activeOpacity={props.disabled ? 1 : 0.7}>
       <StyledImage source={image} resizeMode="contain" />
     </StarWrapper>
   );
 }
 
 function Rating(props: Props): React.ReactElement {
-  const [value, setValue] = useState(0);
-  const [stars, setStars] = useState(new Array(props.total).fill(false));
-  const active = props.onChange;
-  const disabled = props.disabled;
-
-  const _onChangeValue = (): void => {
-    const result = stars.fill(false);
-    let i = 0;
-    while (i < value) {
-      result[i] = true;
-      i++;
-    }
-
-    setStars(result);
-
-    props.onChange && props.onChange({ value });
+  const _handlePress = (position: number): void => {
+    props.onChange && props.onChange({ value: position + 1 });
```

Comment:
If you change your `onChange` prop to `onChnage: (value: number) => void`, 
`onChange(position + 1)` will be match your type predifined.